extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/CacheInterceptor.java

```diff
@@ -138,6 +146,30 @@ protected Object getCacheKey(Cache cache, List<Short> cacheKeyParameterPositions
         }
     }
 
+    private Object generateKey(Class<? extends CacheKeyGenerator> keyGeneratorClass, Method method,
+            Object[] methodParameterValues) {
+        InstanceHandle<? extends CacheKeyGenerator> keyGeneratorHandle = Arc.container().instance(keyGeneratorClass);
```

Comment:
You can `@Inject Instance<CacheKeyGenerator>` directly in the interceptor and then use `instance.select(keyGeneratorClass)` and `Instance#isResolvable()` and `Instance#get()` instead. This way you would not need to produce an `UnremovableBeanBuildItem` for all the `CacheKeyGenerator` implementations and moreoever `@Dependent` scoped impls would be correctly bound the lifecycle of the interceptor.

Reply:
I will still need to call `Instance#destroy` on the keygen instance after I'm done generating the cache key, right? I just tested your suggestion (which is great, thanks!) without calling `destroy` locally and the instances of `CacheKeyGenerator` are piling up in the memory.

Reply:
Yes, for `@Dependent`  keygens you'd  need to call `destroy()` or even `@Inject io.quarkus.arc.InjectableInstance<CacheKeyGenerator>` and use `getHandle()` to obtain an `io.quarkus.arc.InstanceHandle` which also extends `AutoCloseable`.